web_src/js/features/repo-editor.js

```diff
@@ -131,7 +122,21 @@ export function initRepoEditor() {
     });
     if ($(this).val()) parts.push($(this).val());
     $('#tree_path').val(parts.join('/'));
-  }).trigger('keyup');
+  });
+
+  $editFilename.on('keyup', function (e) {
+    const $section = $('.breadcrumb span.section');
+    const $divider = $('.breadcrumb div.divider');
+    let value;
+
+    if (e.keyCode === 8 && getCursorPosition($(this)) === 0 && $section.length > 0) {
+      value = $section.last().find('a').text();
+      $(this).val(value + $(this).val());
+      $(this)[0].setSelectionRange(value.length, value.length);
+      $section.last().remove();
+      $divider.last().remove();
+    }
```

Comment:
Does it also need to update `$('#tree_path').val(...);` like above? It could be a common local function to "join" the "full" parts and update the `$('#tree_path')`